jwst/persistence/tests/test_persistence.py

```diff
@@ -0,0 +1,268 @@
+    """Class to handle testing trap pars."""
+
+    def test_get_parameters(self):
+        # Create empty DataSet
+        ds = persistence.DataSet(None, None, 40.0, False, None, None, None)
+        ds.trappars_model = create_trappars_model()
+        pars = ds.get_parameters()
+        np.testing.assert_equal(pars[0], np.array([180.0, 270.0, 80.0]))
+        np.testing.assert_equal(pars[1], np.array([-0.0004, -0.004, -0.0009]))
+        np.testing.assert_equal(pars[2], np.array([290.0, 140.0, 320.0]))
+        np.testing.assert_equal(pars[3], np.array([-0.01, -0.001, -0.0002]))
+
+    def test_get_capture_param(self, generate_trap_pars):
+        """
+        Test the get_capture_param method.
+
+        Input data is a tuple of 4 float arrays of length 3
+        But still need to create an empty dataset
+        """
+        ds = persistence.DataSet(None, None, 40.0, False, None, None, None)
+        pars = generate_trap_pars()
+        index = 1
+        returned_pars = ds.get_capture_param(pars, index)
+        assert returned_pars[0] == 270.0
+        assert returned_pars[1] == -0.004
+        assert returned_pars[2] == 140.0
+
+    def test_get_decay_param(self, generate_trap_pars):
+        ds = persistence.DataSet(None, None, 40.0, False, None, None, None)
+        pars = generate_trap_pars()
+        index = 1
+        capture_param = ds.get_capture_param(pars, index)
+        assert capture_param == -0.001
+
+
+def test_get_group_info(create_sci_model):
+    output_model = create_sci_model(3, 12, 512, 512, 257, 769)
+    ds = persistence.DataSet(output_model, None, 40.0, False, None, None, None)
+    integration = 1
+    ds.get_group_info(integration)
+    assert ds.tframe == 10.73677
+    assert ds.tgroup == 21.47354
+    assert ds.ngroups == 12
+    assert ds.nframes == 1
+    assert ds.groupgap == 1
+    assert ds.nresets == 1
+
+
+def create_persistencesat_model():
+    scidata = np.ones((2048, 2048), dtype=np.float32)
+    scidata = scidata[:] * 40.0
+    persat_model = datamodels.PersistenceSatModel(data=scidata)
+    persat_model.meta.subarray.xstart = 1
+    persat_model.meta.subarray.ystart = 1
+    return persat_model
+
+
+def test_do_all(create_sci_model, create_traps_filled_model, create_trap_density_model):
+    """
+    Test the do_all method.
+
+    Needs a Dataset object with:
+        .output_obj.data.shape (which must have len 4, like a RampModel)
+        .output_obj.meta.exposure.group_time
+        .trap_density.data.shape
+        .traps_filled (will create one if None)
+        .trap_density
+        .persistencesat
+        .output_obj.meta.exposure.start_time
+        .output_obj.meta.exposure.end_time
+        .traps_filled.data
+        .save_persistence
+    """
+    input_model = create_sci_model(3, 12, 512, 512, 257, 769)
+    traps_filled_model = create_traps_filled_model(2048, 2048)
+    trappars_model = create_trappars_model()
+    trap_density_model = create_trap_density_model(2048, 2048)
```

Comment:
How long does this test take to run?  Do these need to be so large?  Could the test still run on something that was like 10x10?

Reply:
The full set of tests takes 1.35s on my machine

Reply:
ok cool then its all good